pandas/core/generic.py

```diff
@@ -3589,13 +3589,20 @@ def _slice(self: FrameOrSeries, slobj: slice, axis=0) -> FrameOrSeries:
         result._set_is_copy(self, copy=is_copy)
         return result
 
-    def _iset_item(self, loc: int, value) -> None:
+    def _iset_item(self, loc: int, value, clear: bool_t = True) -> None:
```

Comment:
where do we not clear?

Reply:
we get here via NDFrame._maybe_cache_changed.

These are pretty spaghetti-ish, looking forward to cleaning them out